ArdusatSDK.cpp

```diff
@@ -453,266 +257,1212 @@ const char * valuesToCSV(const char *sensorName, unsigned long timestamp, int nu
+
+/**
+ * @brief   Takes a reading from the sensor and returns value in JSON format
+ * @ingroup orientation
+ * @param   sensorName The text to display next to the value
+ * @param   accel Existing Acceleration Sensor to use
+ * @param   mag Existing Magnetic Sensor to use
+ */
+const char * Orientation::readToJSON(Acceleration & acc, Magnetic & mag, const char * sensorName) {
+  Orientation::read(acc, mag);
+  return Orientation::toJSON(sensorName);
+}
+
+/**
+ * @brief   Returns last read value in CSV format
+ * @ingroup orientation
+ * @param   sensorName The text to display next to the value
+ */
+const char * Orientation::toCSV(const char * sensorName) {
+  if (Orientation::initialized)
+    return valuesToCSV(sensorName, Orientation::header.timestamp, 3,
+                       Orientation::roll, Orientation::pitch, Orientation::heading);
+  else
+    return "";
+}
+
+/**
+ * @brief   Returns last read value in JSON format
+ * @ingroup orientation
+ * @param   sensorName The text to display next to the value
+ */
+const char * Orientation::toJSON(const char * sensorName) {
+  if (Orientation::initialized)
+    return _tripleValueToJSON(sensorName, 8, Orientation::header.unit, "%sRoll", Orientation::roll,
+                             "%sPitch", Orientation::pitch, "%sHeading", Orientation::heading);
+  else
+    return "";
+}
+
+
+/**************************************************************************//**
+ * @brief   Constructs Pressure object
+ * @ingroup pressure
+ *
+ * Example Usage:
+ * @code
+ *     Pressure press = Pressure();        // Instantiate sensor object
+ *     press.begin();                      // Initialize sensor
+ *     Serial.println(press.readToJSON()); // Read and print values in JSON
+ * @endcode
+ *****************************************************************************/
+Pressure::Pressure(void) {
+  Pressure::sensorId = SENSORID_BMP180;
+  Pressure::bmp085_mode = BMP085_MODE_ULTRAHIGHRES;
+  Pressure::header.unit = DATA_UNIT_HECTOPASCAL;
+  Pressure::initialized = false;
+}
+
+Pressure::~Pressure() {
+}
+
+/**
+ * @brief   Initializes the sensor with basic configurations
+ * @ingroup pressure
+ *
+ * @retval true  Successfully initialized
+ * @retval false Failed to initialize
+ */
+boolean Pressure::begin(void) {
+  return Pressure::begin(Pressure::bmp085_mode);
+}
+
+/**
+ * @brief   Initializes the sensor with advanced configurations
+ * @ingroup pressure
+ *
+ * @param mode Advanced configuration for BMP180's resolution mode
+ *     - BMP085_MODE_ULTRALOWPOWER
+ *     - BMP085_MODE_STANDARD
+ *     - BMP085_MODE_HIGHRES
+ *     - BMP085_MODE_ULTRAHIGHRES (Default)
+ *
+ * @retval true  Successfully initialized
+ * @retval false Failed to initialize
+ */
+boolean Pressure::begin(bmp085_mode_t mode) {
+  Pressure::bmp085_mode = mode;
+  Pressure::initialized = start_sensor_or_err(pressure_sensor_name, bmp180_init(mode));
+
+  catchSpaceboard();
+  if (ARDUSAT_SPACEBOARD && !Pressure::initialized) {
```

Comment:
use the shared version of this functionality (assuming this will be a redundant comment after the refactors suggested above)